Electron/src/hooks/useSongChangeContextApi.tsx

```diff
@@ -0,0 +1,45 @@
+import { createContext, useContext, useState, ReactNode } from 'react';
+import Global from 'global/global';
+
+// type of context state
+interface SongNameContextType {
+  songName: string;
+  changeSongName: (newSongName: string) => void;
+}
+
+// Create the context for songname and change song name
+const SongNameContext = createContext<SongNameContextType | undefined>(
+  undefined,
+);
+
+// Define the provider component
+export function SongNameChangeContextProvider({
+  children,
+}: {
+  children: ReactNode;
+}) {
+  const [songName, setSongName] = useState<string>(Global.noSongPlaying);
+
+  const changeSongName = (newSongName: string) => {
+    setSongName(newSongName);
+  };
+
+  return (
+    <SongNameContext.Provider value={{ songName, changeSongName }}>
+      {children}
+    </SongNameContext.Provider>
+  );
+}
```

Comment:
@AntonioMrtz value object may create new reference at every re-render of provider so should we use useMemo

Reply:
Yeah @SaurabhGurde , no problem with that, add the useMemo behaviour